src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageSearch/PackageSearchResultRendererTable.cs

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+using NuGet.Configuration;
+using NuGet.Protocol.Core.Types;
+
+namespace NuGet.CommandLine.XPlat
+{
+    internal class PackageSearchResultRendererTable : IPackageSearchResultRenderer
+    {
+        private PackageSearchArgs _args;
+        private const int LineSeparatorLength = 40;
+        private static readonly string SourceSeparator = new('*', LineSeparatorLength);
+
+        public PackageSearchResultRendererTable(PackageSearchArgs packageSearchArgs)
+        {
+            _args = packageSearchArgs;
+        }
+
+        public async Task Add(PackageSource source, Task<IEnumerable<IPackageSearchMetadata>> completedSearchTask)
```

Comment:
Can we add the IEnumerable instead of the task?

Reply:
That would mean I would have to await for the search task which would involve checking whether the task is null or not. If it is null I would have to log the fact that it was not able to obtain a search resource from the source.  In addition, there is also more catching and logging errors needed to be done based on the response of the task. I was thinking, by doing all of this in the renderer side, I would guarantee all the rendering is done by the renderer. And PackageSearchRunner would be only responsible for constructing these search tasks and no logging.

Reply:
Now the renderer is being responsible for ensuring the task completes, which you can argue is goes beyond it's responsibility. 

Take https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.CommandLine.XPlat/ListPackage/IReportRenderer.cs for example, it even has an AddProblem option. 

Curious what @NuGet/nuget-client thinks about that.

Reply:
That makes sense. I am thinking I could add 
```
        void Add(PackageSource source, IEnumerable<IPackageSearchMetadata> completedSearch);
        void Add(PackageSource source, string error);
```
two add statements to the renderer rather and do  all the other tasks in the search runner

Reply:
Fixed